<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>682</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>500</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>SavvyCAN</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
    <item>
     <widget class="QTableView" name="canFramesView">
      <property name="font">
       <font>
        <pointsize>11</pointsize>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="verticalScrollMode">
       <enum>QAbstractItemView::ScrollPerPixel</enum>
      </property>
      <property name="wordWrap">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="lblRemoteConn">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Remote Connection Active:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineRemoteKey">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>1000</width>
          <height>50</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>150</width>
          <height>50</height>
         </size>
        </property>
        <property name="font">
         <font>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="maxLength">
         <number>8</number>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblContMsg">
        <property name="palette">
         <palette>
          <active>
           <colorrole role="WindowText">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>255</red>
              <green>0</green>
              <blue>0</blue>
             </color>
            </brush>
           </colorrole>
          </active>
          <inactive>
           <colorrole role="WindowText">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>255</red>
              <green>0</green>
              <blue>0</blue>
             </color>
            </brush>
           </colorrole>
          </inactive>
          <disabled>
           <colorrole role="WindowText">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>120</red>
              <green>120</green>
              <blue>120</blue>
             </color>
            </brush>
           </colorrole>
          </disabled>
         </palette>
        </property>
        <property name="font">
         <font>
          <pointsize>11</pointsize>
          <weight>75</weight>
          <italic>true</italic>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Total Frames Captured:</string>
        </property>
       </widget>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="lbNumFrames">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="label_6">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Frames Per Second:</string>
        </property>
       </widget>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="lbFPS">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnCaptureToggle">
        <property name="text">
         <string>Suspend Capturing</string>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnNormalize">
        <property name="text">
         <string>Normalize Frame Timing</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnClearFrames">
        <property name="text">
         <string>Clear Frames</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbAutoScroll">
        <property name="text">
         <string>Auto Scroll Window</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbOverwrite">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Overwrite Mode</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbInterpret">
        <property name="text">
         <string>Interpret Frames</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnExpandAll">
        <property name="text">
         <string>Expand All Rows</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnCollapseAll">
        <property name="text">
         <string>Collapse All Rows</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Bus Filtering:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="listBusFilters">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>175</width>
          <height>40</height>
         </size>
        </property>
        <property name="flow">
         <enum>QListView::LeftToRight</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Frame Filtering:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="listFilters">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>160</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>175</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QPushButton" name="btnFilterAll">
          <property name="text">
           <string>All</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnFilterNone">
          <property name="text">
           <string>None</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_RE_Tools">
    <property name="title">
     <string>RE Tools</string>
    </property>
    <addaction name="actionFlow_View"/>
    <addaction name="actionGraph_Dta"/>
    <addaction name="actionFrame_Data_Analysis"/>
    <addaction name="actionFile_Comparison"/>
    <addaction name="actionDBC_Comparison"/>
    <addaction name="actionRange_State_2"/>
    <addaction name="actionSingle_Multi_State_2"/>
    <addaction name="actionISO_TP_Decoder"/>
    <addaction name="actionSniffer"/>
    <addaction name="actionCapture_Bisector"/>
    <addaction name="actionSignal_Viewer"/>
    <addaction name="actionTemporal_Graph"/>
   </widget>
   <widget class="QMenu" name="menuSend_Frames">
    <property name="title">
     <string>Send Frames</string>
    </property>
    <addaction name="action_Playback"/>
    <addaction name="action_Custom"/>
    <addaction name="actionScripting_INterface"/>
    <addaction name="actionFuzzing"/>
    <addaction name="actionUDS_Scanner"/>
    <addaction name="actionFirmware_Update"/>
    <addaction name="actionMotorControlConfig"/>
   </widget>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_Log_File"/>
    <addaction name="actionSave_Filtered_Log_File"/>
    <addaction name="actionSave_Log_File"/>
    <addaction name="actionSave_Continuous_Logfile"/>
    <addaction name="separator"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_Filter_Definition"/>
    <addaction name="actionSave_Filter_Definition"/>
    <addaction name="separator"/>
    <addaction name="actionDBC_File_Manager"/>
    <addaction name="actionSave_Decoded_Frames"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
    <addaction name="separator"/>
    <addaction name="actionExit_Application"/>
   </widget>
   <widget class="QMenu" name="menuConnection">
    <property name="title">
     <string>Connection</string>
    </property>
    <addaction name="actionSetup"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menu_RE_Tools"/>
   <addaction name="menuSend_Frames"/>
   <addaction name="menuConnection"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_Playback">
   <property name="text">
    <string>Playback</string>
   </property>
  </action>
  <action name="action_Custom">
   <property name="text">
    <string>Custom</string>
   </property>
  </action>
  <action name="actionFlow_View">
   <property name="text">
    <string>Flow View</string>
   </property>
  </action>
  <action name="actionFuzzy_Scope">
   <property name="text">
    <string>Fuzzy Scope</string>
   </property>
  </action>
  <action name="actionGraph_Dta">
   <property name="text">
    <string>New Graphing Window</string>
   </property>
  </action>
  <action name="actionOpen_Log_File">
   <property name="text">
    <string>Load Log File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave_Log_File">
   <property name="text">
    <string>Save Log File</string>
   </property>
  </action>
  <action name="actionFrame_Data_Analysis">
   <property name="text">
    <string>Frame Data Analysis</string>
   </property>
  </action>
  <action name="actionLoad_DBC_File">
   <property name="text">
    <string>Load DBC File</string>
   </property>
  </action>
  <action name="actionEdit_Messages_Signals">
   <property name="text">
    <string>Edit Messages/Signals</string>
   </property>
  </action>
  <action name="actionSave_DBC_File">
   <property name="text">
    <string>Save DBC File</string>
   </property>
  </action>
  <action name="actionSave_Decoded_Frames">
   <property name="text">
    <string>Save Decoded Frames</string>
   </property>
  </action>
  <action name="actionExit_Application">
   <property name="text">
    <string>Exit Application</string>
   </property>
  </action>
  <action name="actionFile_Comparison">
   <property name="text">
    <string>File Comparison</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>Preferences</string>
   </property>
  </action>
  <action name="actionSave_Filtered_Log_File">
   <property name="text">
    <string>Save Filtered Log File</string>
   </property>
  </action>
  <action name="actionLoad_Filter_Definition">
   <property name="text">
    <string>Load Filter Definition</string>
   </property>
  </action>
  <action name="actionSave_Filter_Definition">
   <property name="text">
    <string>Save Filter Definition</string>
   </property>
  </action>
  <action name="actionFirmware_Uploader">
   <property name="text">
    <string>Firmware Uploader</string>
   </property>
  </action>
  <action name="actionSetup">
   <property name="text">
    <string>Open Connection Window</string>
   </property>
  </action>
  <action name="actionConnect">
   <property name="text">
    <string>Connect</string>
   </property>
  </action>
  <action name="actionScripting_INterface">
   <property name="text">
    <string>Scripting Interface</string>
   </property>
  </action>
  <action name="actionRange_State_2">
   <property name="text">
    <string>Range State</string>
   </property>
  </action>
  <action name="actionSingle_Multi_State_2">
   <property name="text">
    <string>Single/Multi State</string>
   </property>
  </action>
  <action name="actionDBC_File_Manager">
   <property name="text">
    <string>DBC File Manager</string>
   </property>
  </action>
  <action name="actionFuzzing">
   <property name="text">
    <string>Fuzzing</string>
   </property>
  </action>
  <action name="actionUDS_Scanner">
   <property name="text">
    <string>UDS Scanner</string>
   </property>
  </action>
  <action name="actionISO_TP_Decoder">
   <property name="text">
    <string>ISO-TP Decoder</string>
   </property>
  </action>
  <action name="actionSniffer">
   <property name="text">
    <string>Sniffer</string>
   </property>
  </action>
  <action name="actionFirmware_Update">
   <property name="text">
    <string>Firmware Update</string>
   </property>
  </action>
  <action name="actionMotorControlConfig">
   <property name="text">
    <string>MotorControlConfig</string>
   </property>
  </action>
  <action name="actionCapture_Bisector">
   <property name="text">
    <string>Capture Bisector</string>
   </property>
  </action>
  <action name="actionSignal_Viewer">
   <property name="text">
    <string>Signal Viewer</string>
   </property>
  </action>
  <action name="actionSave_Continuous_Logfile">
   <property name="text">
    <string>Start Continuous Logging</string>
   </property>
  </action>
  <action name="actionTemporal_Graph">
   <property name="text">
    <string>Temporal Graph</string>
   </property>
  </action>
  <action name="actionDBC_Comparison">
   <property name="text">
    <string>DBC Comparison</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
 <slots>
  <slot>handleLoadFile()</slot>
  <slot>connButtonPress()</slot>
  <slot>showGraphingWindow()</slot>
  <slot>showFrameDataAnalysis()</slot>
 </slots>
</ui>
